File: ai-go/models/meta_attr.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type MetaAttr struct {
	ID uint64 `gorm:"primaryKey" json:"id"`
	// EventID   uint64    `gorm:"not null" json:"event_id"`
	AttrName  string `gorm:"size:200;not null" json:"attr_name"`
	AttrType  string `gorm:"size:50;not null;default:'string'" json:"attr_type"`
	AttrDesc  string `gorm:"type:text" json:"attr_desc"`
	IsDeleted bool   `gorm:"not null;default:false" json:"is_deleted"`
	// CreatedAt time.Time `gorm:"not null;default:CURRENT_TIMESTAMP" json:"-"`
	// UpdatedAt time.Time `gorm:"not null;default:CURRENT_TIMESTAMP" json:"-"`
	CreatedAt time.Time `gorm:"type:datetime;not null;default:CURRENT_TIMESTAMP" json:"-"`
	UpdatedAt time.Time `gorm:"type:datetime;not null;default:CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP" json:"-"`
	// Events    []MetaEvent `gorm:"many2many:meta_relation;" json:"events,omitempty"`
	Events     []MetaEvent `gorm:"many2many:meta_relation;foreignKey:ID;joinForeignKey:AttrID;References:ID;joinReferences:EventID" json:"events,omitempty"`
	CreatedStr string      `gorm:"-" json:"created_at"`
	UpdatedStr string      `gorm:"-" json:"updated_at"`
}

func (m *MetaAttr) AfterFind(*gorm.DB) error {
	m.CreatedStr = m.CreatedAt.Format("2006-01-02 15:04:05")
	m.UpdatedStr = m.UpdatedAt.Format("2006-01-02 15:04:05")
	return nil
}

func (MetaAttr) TableName() string {
	return "meta_attr"
}

File: ai-java/src/main/java/com/example/entity/MetaAttr.java
package com.example.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

@Data
@Entity
@Table(name = "meta_attr")
@SQLDelete(sql = "UPDATE meta_attr SET is_deleted = true WHERE id = ?")
@Where(clause = "is_deleted = false")
public class MetaAttr {
    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "attr_name", length = 200, nullable = false)
    private String attrName;

    @Column(name = "attr_type", length = 50, nullable = false)
    private String attrType = "string";

    @Column(name = "attr_desc", columnDefinition = "TEXT")
    private String attrDesc;

    @Column(name = "is_deleted", nullable = false)
    private Boolean isDeleted = false;

    @JsonIgnore
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @JsonIgnore
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @JsonIgnore
    @ManyToMany(mappedBy = "attributes")
    private List<MetaEvent> events;

    @Transient
    private String createdStr;

    @Transient
    private String updatedStr;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    @PostLoad
    protected void onLoad() {
        if (createdAt != null) {
            createdStr = createdAt.format(DATE_FORMATTER);
        }
        if (updatedAt != null) {
            updatedStr = updatedAt.format(DATE_FORMATTER);
        }
    }
} 